spec/java/lang/System.StdOut.lsl

```diff
@@ -0,0 +1,348 @@
+
+import java/lang/System;
+import runtime/utils/VoidOutputStream;
+
+
+// automata
+
+automaton System_PrintStreamAutomaton
+(
+)
+: System_PrintStream
+{
+    // states and shifts
+
+    initstate Initialized;
+
+    shift Initialized -> self by [
+        // instance methods
+        append (System_PrintStream, CharSequence),
+        append (System_PrintStream, CharSequence, int, int),
+        append (System_PrintStream, char),
+        checkError,
+        close,
+        flush,
+        format (System_PrintStream, Locale, String, array<Object>),
+        format (System_PrintStream, String, array<Object>),
+        print (System_PrintStream, Object),
+        print (System_PrintStream, String),
+        print (System_PrintStream, boolean),
+        print (System_PrintStream, char),
+        print (System_PrintStream, array<char>),
+        print (System_PrintStream, double),
+        print (System_PrintStream, float),
+        print (System_PrintStream, int),
+        print (System_PrintStream, long),
+        printf (System_PrintStream, Locale, String, array<Object>),
+        printf (System_PrintStream, String, array<Object>),
+        println (System_PrintStream),
+        println (System_PrintStream, Object),
+        println (System_PrintStream, String),
+        println (System_PrintStream, boolean),
+        println (System_PrintStream, char),
+        println (System_PrintStream, array<char>),
+        println (System_PrintStream, double),
+        println (System_PrintStream, float),
+        println (System_PrintStream, int),
+        println (System_PrintStream, long),
+        write (System_PrintStream, array<byte>),
+        write (System_PrintStream, array<byte>, int, int),
+        write (System_PrintStream, int),
+    ];
+
+    // internal variables
+
+    var closed: boolean = false;
+    var error: boolean = false;
+
+
+    // utilities
+
+    @AutoInline @Phantom proc _throwNPE (): void
+    {
+        action THROW_NEW("java.lang.NullPointerException", []);
+    }
+
+
+    @AutoInline @Phantom proc _checkOpen (): void
+    {
+        if (this.closed)
+            this.error = true;
+    }
+
+
+    // constructors
+
+    // static methods
+
+    // methods
+
+    fun *.append (@target self: System_PrintStream, csq: CharSequence): PrintStream
+    {
+        if (csq == null)
+            _throwNPE();
+
+        _checkOpen();
+
+        result = self as Object as PrintStream;
+    }
+
+
+    fun *.append (@target self: System_PrintStream, csq: CharSequence, start: int, end: int): PrintStream
+    {
+        if (csq == null)
+            _throwNPE();
+
+        val size: int = action CALL_METHOD(csq, "length", []);
+        if (start < 0 || end >= size)
+            action THROW_NEW("java.lang.IndexOutOfBoundsException", []);
+
+        _checkOpen();
+
+        result = self as Object as PrintStream;
+    }
+
+
+    fun *.append (@target self: System_PrintStream, c: char): PrintStream
+    {
+        _checkOpen();
```

Comment:
This overall implementation assumes that everything is being written into nothingness, never being received back.
So it is effectively the same to do nothing at all with the data being send out.

And, yes, it is *technically* possible to construct a program (or rather, a pair of programs) that writes a piece of text into the `stdout` and then reads it back via `stdin` expecting it to be exactly the same and crashing otherwise.

To (partially) eliminate such cases, `System.in` is being modeled via `SymbolicInputStream` which is capable of generating arbitrary strings of data, making it possible to explore both cases: *"the same"* and *"not the same"*.